### PR TITLE
meli: 0.8.7 -> 0.8.8

### PR DESCRIPTION
**Pull request by ehamberg**
Release notes: [git.meli-email.org/meli/meli/releases/tag/v0.8.8](https://git.meli-email.org/meli/meli/releases/tag/v0.8.8)

~~Building on Darwin works, but there is a test-failure that seems to be Darwin-specific, so holding off adding Darwin to the list of `platforms` for now.~~

Building on Darwin works again!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by toastal — CHANGES_REQUESTED** [review]

File: pkgs/by-name/me/meli/package.nix [hunk 1]
```diff
@@ -35,7 +35,7 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-2+JIehi2wuWdARbhFPvNPIJ9ucZKWjNSORszEG9lyjw=";
```
[the remark this answers is not in the document]
Comment (ehamberg) [reply pkgs/by-name/me/meli/package.nix]: D'oh! Yes. Updated.

**Review by toastal — APPROVED** [review]
LGTM

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project. I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

**Comment by ehamberg** [comment]
The following test fails on Darwin, so I removed Darwin from the `platforms` list again for now, since a test failure in fd locking sounds a bit scary.
```
---- utils::tests::test_fd_locks stdout ----
thread 'utils::tests::test_fd_locks' panicked at melib/src/utils/tests.rs:309:10:
called `Result::unwrap_err()` on an `Ok` value: FileLock(File { fd: 11, path: "/private/tmp/nix-build-meli-0.8.8.drv-0/.tmpbQ6yBc/test.txt", read: true, write: true })
```

**Comment by ehamberg** [comment]
Update: This was quickly fixed by the author in https://git.meli-email.org/meli/meli/pulls/524, so Darwin is back in the `platforms` list :)

**Comment by epilys** [comment]
Hi all, out of curiosity, is there a technical issue with the `aarch64-linux` build? Is there something I can do about it?

**Comment by teto** [comment]
passed a nixpkgs-review + had toastal approval (who does't have merge rights so I took it upon me to merge). Nice changelog ! 

> Hi all, out of curiosity, is there a technical issue with the aarch64-linux build? 

not all builders are equal. aarch64 and darwin usually perform worse than linux builders. Currently the ofborg eval job is especially slow which is a problem but it's not like that usually.

**Comment by epilys** [comment]
Thanks everybody, I appreciate your efforts for a humble package like this one :)

**Comment by epilys** [comment]
@teto hey Matthieu, this is a CLI tool to sync stuff externally of meli, correct? Can you file a bug report on the meli issue tracker (you can log in with your github account) and I will look at it ASAP.

**Comment by teto** [comment]
https://github.com/elizagamedev/mujmap is a cool project to sync jmap mail but the maintainer is not very active. I can deal with that myself dont worry.

I postponed my usage of meli because the output was messed up with kitty or wezterm (I use kitty). By experience when I added guicursor terminal support in neovim, I know it's tricky to get things right with terminals but that would be very cool to have proper output in kitty. All my meli issues are closed but this one references kitty issues as well https://git.meli-email.org/meli/meli/issues/388

**Comment by epilys** [comment]
@teto I'm sorry about your experience :( FWIW I don't get those problems in kitty or wezterm, so I am obviously unsuccessfully setting up the environment on my end. I would certainly appreciate help in that area since I'm only one person and it's difficult to solve bugs across different use setups (that is the reason solving these kinds of bugs takes longer than usual, unfortunately)

Thanks for letting me know!

**Pull request merged**